bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/CMakeBuildTab2.java

```diff
@@ -340,4 +352,25 @@ public String getName()
 		return "CMake"; //$NON-NLS-1$
 	}
 
+	private String getCmakeArgumentsWithAbsProjectPath(IProject project, String cmakeArgumets)
+	{
+		String[] cmakeArgsArr = cmakeArgsText.getText().trim().replaceAll(" +", " ").split(" "); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+		String buildFolder = StringUtil.EMPTY;
+		for (int i = 0; i < cmakeArgsArr.length; i++)
+		{
+			if (cmakeArgsArr[i].equals("-B") && i + 1 < cmakeArgsArr.length) //$NON-NLS-1$
+			{
+				buildFolder = cmakeArgsArr[i + 1];
+				break;
+			}
+		}
+
+		if (!Path.of(buildFolder).isAbsolute())
+		{
+			cmakeArgumets = cmakeArgumets.replaceFirst("(?<=-B)\\s+(\\S+)", //$NON-NLS-1$
```

Comment:
Add comment about regex what it means, that improves code readability

Reply:
done